x/ownership/internal/keeper/keeper.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/mesg-foundation/engine/hash"
+	"github.com/mesg-foundation/engine/ownership"
+	"github.com/mesg-foundation/engine/x/ownership/internal/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// Keeper of the ownership store
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      *codec.Codec
+}
+
+// NewKeeper creates a ownership keeper
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey) Keeper {
+	keeper := Keeper{
+		storeKey: key,
+		cdc:      cdc,
+	}
+	return keeper
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+// List returns all ownerships.
+func (k *Keeper) List(ctx sdk.Context) ([]*ownership.Ownership, error) {
+	var (
+		ownerships []*ownership.Ownership
+		iter       = ctx.KVStore(k.storeKey).Iterator(nil, nil)
+	)
+	defer iter.Close()
+
+	for iter.Valid() {
+		var o *ownership.Ownership
+		if err := k.cdc.UnmarshalBinaryLengthPrefixed(iter.Value(), &o); err != nil {
+			return nil, err
+		}
+		ownerships = append(ownerships, o)
+		iter.Next()
+	}
+	return ownerships, nil
+}
+
+// Set creates a new ownership.
+func (k Keeper) Set(ctx sdk.Context, owner sdk.AccAddress, resourceHash hash.Hash, resource ownership.Ownership_Resource) (*ownership.Ownership, error) {
+	store := ctx.KVStore(k.storeKey)
+	hashes := k.findOwnerships(store, "", resourceHash)
+	if len(hashes) > 0 {
+		return nil, fmt.Errorf("resource %s:%q already has an owner", resource, resourceHash)
+	}
+
+	ownership := &ownership.Ownership{
+		Owner:        owner.String(),
+		Resource:     resource,
+		ResourceHash: resourceHash,
+	}
+	ownership.Hash = hash.Dump(ownership)
+
+	store.Set(ownership.Hash, k.cdc.MustMarshalBinaryLengthPrefixed(ownership))
```

Comment:
Let's avoid to use `MustXXX` functions and return the error.

Reply:
This one is from the template. I think it's good to have `Must` as this will only panic if the concreate type is not register (same with json).

Reply:
but this function already return an error.. why not using it?